x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/AutoFollowCoordinator.java

```diff
@@ -460,6 +467,56 @@ static String getFollowerIndexName(AutoFollowPattern autoFollowPattern, String l
             };
         }
 
+        void cleanFollowedLeaderIndices(final ClusterState remoteClusterState,
+                                                final List<String> patterns) {
+            updateAutoFollowMetadata(cleanFollowedLeaderIndices(remoteClusterState.metaData(), patterns), e -> {
+                if (e != null) {
+                    LOGGER.warn("Error occured while cleaning followed leader indices", e);
+                }
+            });
+        }
+
+        static Function<ClusterState, ClusterState> cleanFollowedLeaderIndices(final MetaData remoteMetadata,
+                                                                               final List<String> autoFollowPatternNames) {
+            return currentState -> {
+                AutoFollowMetadata currentAutoFollowMetadata = currentState.metaData().custom(AutoFollowMetadata.TYPE);
+                Map<String, List<String>> newFollowedIndexUUIDS = new HashMap<>(currentAutoFollowMetadata.getFollowedLeaderIndexUUIDs());
```

Comment:
I think this needs a rename. I got confused by it a few times later on in the code. Perhaps `autoFollowPatternNameToFollowedIndexUUIDs`?